overlord/snapstate/handlers.go

```diff
@@ -324,19 +324,28 @@ func (m *SnapManager) installOneBaseOrRequired(t *state.Task, snapName string, c
 		return nil, err
 	}
 
-	inProgress := func(snapName string) (bool, error) {
+	shouldAttemptInstall := func(snapName string) (bool, error) {
 		linkTask, err := findLinkSnapTaskForSnap(st, snapName)
 		if err != nil {
 			return false, err
 		}
 
 		if linkTask == nil {
 			// snap is not being installed
+			return true, nil
+		}
+
+		// if we are remodeling, then we should return early due to the way that
+		// tasks are ordered by the remodeling code. specifically, all snap
+		// downloads during a remodel happen prior to snap installation. thus,
+		// we cannot wait for snaps to be installed here. see remodelTasks for
+		// more information on how the tasks are ordered.
+		if deviceCtx.ForRemodeling() {
 			return false, nil
 		}
 
 		// snap is being installed, retry later
```

Comment:
is this comment making sense in all cases now?

Reply:
Updated to account for the refresh case